### PR TITLE
Azure auth: do not use CLI provider by default when running in a cloud service

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
This seems to be causing some issues for customers, and in a cloud service the CLI credentials would not work anyways.

### CODE REVIEW

**Comment by berndverst** [comment]
This is a breaking change sorry. I personally rely on this OSS behavior. So we should not merge this.

**Review by berndverst — CHANGES_REQUESTED** [review]
This is a breaking change IMO.

**Comment by ItalyPaleAle** [comment]
> This is a breaking change IMO.

Please look at the diff. This only impacts when running in a cloud service. outside of a cloud service, CLI credentials still work

**Review by berndverst — APPROVED** [review]

File: common/authentication/azure/auth.go [hunk 1]
```diff
@@ -499,6 +483,23 @@ func (s EnvironmentSettings) GetEnvironment(key string) (val string, ok bool) {
 	return metadata.GetMetadataProperty(s.Metadata, MetadataKeys[key]...)
 }
 
+// Returns true if the application is running on a cloud service with Managed Identity, including: Azure App Service, Azure Functions, Azure Service Fabric, Azure Container Apps, Azure Arc, Azure Cloud Shell.
```
Comment (berndverst) [inline common/authentication/azure/auth.go:486]: You now eliminated support for services powered by IMDS but which uses the default IMDS endpoint.

**Review by berndverst — CHANGES_REQUESTED** [review]

**Review by berndverst — APPROVED** [review]

File: common/authentication/azure/auth.go (hunk 1 above)
Comment (ItalyPaleAle) [reply common/authentication/azure/auth.go:486]: No there are 2 separate functions

**Pull request merged**